### PR TITLE
style(app, components): create alias components for buttons, link, modal

### PR DESCRIPTION
**Pull request by smb2268**
re #9401

# Overview

Add aliased components that incorporate some of Mel's styling updates for ExternalLink, Modal, and Primary, Secondary, Tertiary and Toggle Buttons

# Changelog

- [ ] Set font, color, size, and add icon to external link
- [ ] Add header to modal with proper styling and close button
- [ ] Added optional `noHeaderStyle` prop to `BaseModal` so we can override the current header styles
- [ ] Add correct colors and border radius to all buttons
- [ ] Fixed line height for a few typography styles

# Review requests

Look over these changes and see if they make sense -- I didn't add tests since these are temporary changes and only affect style, so the actual component tests should remain valid for all functionality. Let me know if you think that's okay! I can add test files if necessary.

# Risk assessment

Low

### CODE REVIEW

File: app/src/atoms/Buttons/index.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    css={TYPOGRAPHY.labelSemiBold}
+    {...props}
+  />
+)
+
+export const PrimaryButton = (props: any): JSX.Element => (
+  <NewPrimaryBtn
+    backgroundColor={COLORS.blue}
+    borderRadius="3px"
+    textTransform="none"
```
Comment (Kadee80) [inline app/src/atoms/Buttons/index.tsx]: text transform should be a const now

File: app/src/atoms/Buttons/index.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    overflow="no-wrap"
+    css={TYPOGRAPHY.labelSemiBold}
+    {...props}
+  />
+)
+
+export const PrimaryButton = (props: any): JSX.Element => (
+  <NewPrimaryBtn
+    backgroundColor={COLORS.blue}
+    borderRadius="3px"
```
Comment (Kadee80) [inline app/src/atoms/Buttons/index.tsx]: should we add a borders.ts?

Comment (smb2268) [reply app/src/atoms/Buttons/index.tsx]: Yeah that could be helpful - we can add these border radius consts and also that blue border below tabs

File: app/src/atoms/Buttons/index.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  SIZE_2,
+  NewPrimaryBtn,
+  NewSecondaryBtn,
+} from '@opentrons/components'
+import { ToggleBtn, ToggleBtnProps } from '../ToggleBtn'
+
+export const TertiaryButton = (props: any): JSX.Element => (
+  <NewPrimaryBtn
+    backgroundColor={COLORS.blue}
+    borderRadius="20px"
```
Comment (jerader) [inline app/src/atoms/Buttons/index.tsx]: should we convert pixels to rem or add this as a borderRadius constant?

File: app/src/atoms/Modal/index.tsx [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+            <Box onClick={props.onClose}>
+              <Icon
+                name={'close'}
+                width={SPACING.spacing4}
+                height={SPACING.spacing4}
+              />
+            </Box>
+          )}
+        </Flex>
+        <Divider width="100%" paddingTop="0" paddingBottom={SPACING.spacing4} />
```
Comment (jerader) [inline app/src/atoms/Modal/index.tsx:45]: does the `paddingTop="0"` do anything?

**Review by b-cooper — APPROVED** [review]
Looks good 🔘 . I agree with the other comments regarding border specific consts and not using px. Once those are addressed this is good to go.

File: app/src/atoms/Buttons/index.tsx (hunk 2 above)
Comment (smb2268) [reply app/src/atoms/Buttons/index.tsx:25]: Talked to Brian about this - going to do it as a follow on after talking to Mel about what these consts should be

File: app/src/atoms/Modal/index.tsx (hunk 4 above)
Comment (smb2268) [reply app/src/atoms/Modal/index.tsx:45]: Yeah it does, it overrides the default padding in the `Divider` component which is 3px

**Pull request merged**